mkt/developers/views.py

```diff
@@ -373,12 +374,32 @@ def content_ratings_edit(request, addon_id, addon):
     }
     request.session.modified = True
 
-    return render(request, 'developers/apps/ratings/ratings_edit.html',
-                  {'addon': addon,
-                   'app_name': get_iarc_app_title(addon),
-                   'form': form,
-                   'company': addon.latest_version.developer_name,
-                   'now': datetime.now().strftime('%Y-%m-%d %H:%M:%S')})
+    ctx = {
+        'addon': addon,
+        'app_name': get_iarc_app_title(addon),
+        'form': form,
+        'company': addon.latest_version.developer_name,
+        'now': datetime.now().strftime('%Y-%m-%d %H:%M:%S')
+    }
+
+    if waffle.switch_is_active('iarc-upgrade'):
+        iarc_request_id = IARCRequestID.objects.filter(app_id=addon)
+        if iarc_request_id.exists():
+            iarc_request_id = iarc_request_id[0]
+            outdated = (datetime.now() - iarc_request_id.created >
+                        timedelta(hours=1))
+            if outdated:
+                # IARC Request ID outdated. Re-create.
```

Comment:
The more I think about this the less I like it.

Fist, since you're checking for existence by passing the app, it's effectively a one-to-one relation, so you could use a `OneToOneField` in the model - that would help guarantee that you won't end up with multiple ids per app because of a race condition.

But, more importantly: I wonder how much trouble silently deleting the token if it's too old in the view will cause in the long run. IARC should answer in less time than that, but...